### PR TITLE
Added Behaviors

### PR DESCRIPTION
**Pull request by AidanSmith**
I added a `BehaviorArbitrator` to drive code with teleop, autonomous, and pit modes.

### CODE REVIEW

File: ZukoAzula2016/src/org/team1540/zukoazula/DriveCode.java [hunk 1]
```diff
@@ -18,7 +22,24 @@
     private static final ExtendedMotor rightFrontCAN = FRC.talonCAN(2), rightBackCAN = FRC.talonCAN(3);
     private static final ExtendedMotor leftFrontCAN = FRC.talonCAN(0), leftBackCAN = FRC.talonCAN(1);
 
+    public static final FloatCell autoLeftMotors = new FloatCell();
+    public static final FloatCell autoRightMotors = new FloatCell();
```
Comment (celskeggs) [inline ZukoAzula2016/src/org/team1540/zukoazula/DriveCode.java]: Perhaps these should be moved to the autonomous code, and referenced from there. Better locality, perhaps?

File: ZukoAzula2016/src/org/team1540/zukoazula/DriveCode.java [hunk 2]
```diff
@@ -18,7 +22,24 @@
...
+    public static final FloatOutput autoAllMotors = autoLeftMotors.combine(autoRightMotors);
```
Comment (celskeggs) [inline ZukoAzula2016/src/org/team1540/zukoazula/DriveCode.java]: This should definitely be in an autonomous class, not in DriveCode.

File: ZukoAzula2016/src/org/team1540/zukoazula/DriveCode.java [hunk 3]
```diff
@@ -26,7 +47,8 @@ public static void setup() throws ExtendedMotorFailureException {
         final FloatOutput leftOut = leftMotors.addRamping(0.1f, FRC.constantPeriodic);
         final FloatOutput rightOut = rightMotors.addRamping(0.1f, FRC.constantPeriodic);
 
-        Drive.extendedTank(driveLeftAxis, driveRightAxis, driveLeftTrigger.minus(driveRightTrigger), leftOut, rightOut);
+        leftInput.send(leftOut);
+        rightInput.send(rightOut);
```
Comment (celskeggs) [inline ZukoAzula2016/src/org/team1540/zukoazula/DriveCode.java]: I recommend inlining a bit to make things more concise.

**Comment by celskeggs** [comment]
@tytynguyen any other comments on this, since you did the first version?

File: ZukoAzula2016/src/org/team1540/zukoazula/DriveCode.java [hunk 4]
```diff
@@ -18,21 +22,26 @@
...
-        final FloatOutput rightOut = rightMotors.addRamping(0.1f, FRC.constantPeriodic);
+    public static void setup() throws ExtendedMotorFailureException {
+        leftInput.attach(teleop, driveLeftAxis.plus(driveLeftTrigger.minus(driveRightTrigger)));
+        rightInput.attach(teleop, driveRightAxis.plus(driveRightTrigger.minus(driveLeftTrigger)));
+        leftInput.attach(pit, FloatInput.zero);
+        rightInput.attach(pit, FloatInput.zero);
 
-        Drive.extendedTank(driveLeftAxis, driveRightAxis, driveLeftTrigger.minus(driveRightTrigger), leftOut, rightOut);
+        leftInput.send(leftFrontCAN.simpleControl(FRC.MOTOR_FORWARD).combine(leftBackCAN.simpleControl(FRC.MOTOR_FORWARD)).addRamping(0.1f, FRC.constantPeriodic));
+        rightInput.send(rightFrontCAN.simpleControl(FRC.MOTOR_REVERSE).combine(rightBackCAN.simpleControl(FRC.MOTOR_REVERSE)).addRamping(0.1f, FRC.constantPeriodic));
```
Comment (celskeggs) [inline ZukoAzula2016/src/org/team1540/zukoazula/DriveCode.java]: Hmm. I think that @tytynguyen was right and I was wrong. Maybe move "a.combine(b)" into its own line.

**Comment by celskeggs** [comment]
Also, have you thoroughly tested this on the Emulator?

**Comment by AidanSmith** [comment]
There isn't much to test. It still works in teleop mode, and when I flip the Pit Mode Enable switch in the Poultry Inspector, the motors stopped.

**Comment by celskeggs** [comment]
Do the triggers work?

**Comment by AidanSmith** [comment]
Oh, wow. Actually they don't.

**Comment by AidanSmith** [comment]
The "drive forward" trigger makes it rotate in place, and the "drive reverse" trigger makes it rotate the other way.

**Comment by celskeggs** [comment]
And that's exactly why I asked. (I was hoping that @tytynguyen would catch this but it's an acquired skill.)

**Comment by AidanSmith** [comment]
And I put @amackworth's emojis back.

File: ZukoAzula2016/src/org/team1540/zukoazula/DriveCode.java [hunk 5]
```diff
@@ -10,29 +14,34 @@
...
+    private static final ArbitratedFloat leftInput = behaviors.addFloat();
+    private static final ArbitratedFloat rightInput = behaviors.addFloat();
+    private static final Behavior teleop = behaviors.addBehavior("Teleop", FRC.inTeleopMode());
+    private static final Behavior pit = behaviors.addBehavior("Pit Mode", ZukoAzula.mainTuning.getBoolean("Pit Mode Enable", false));
 
-        final FloatOutput leftOut = leftMotors.addRamping(0.1f, FRC.constantPeriodic);
-        final FloatOutput rightOut = rightMotors.addRamping(0.1f, FRC.constantPeriodic);
+    public static void setup() throws ExtendedMotorFailureException {
+        leftInput.attach(teleop, driveLeftAxis.minus(driveLeftTrigger.minus(driveRightTrigger)));
+        rightInput.attach(teleop, driveRightAxis.plus(driveRightTrigger.minus(driveLeftTrigger)));
```
Comment (celskeggs) [inline ZukoAzula2016/src/org/team1540/zukoazula/DriveCode.java:35]: Uh... you could just make them symmetrical? It would be much more clear.

**Comment by celskeggs** [comment]
Did you see my earlier comment about potentially moving the "a.combine(b)" expressions into their own lines?

**Comment by AidanSmith** [comment]
I did not. (But I do now.)

File: ZukoAzula2016/src/org/team1540/zukoazula/DriveCode.java [hunk 6]
```diff
@@ -10,29 +14,37 @@
...
 
     private static final ExtendedMotor rightFrontCAN = FRC.talonCAN(2), rightBackCAN = FRC.talonCAN(3);
     private static final ExtendedMotor leftFrontCAN = FRC.talonCAN(0), leftBackCAN = FRC.talonCAN(1);
 
-    public static void setup() throws ExtendedMotorFailureException {
+    private static final BehaviorArbitrator behaviors = new BehaviorArbitrator("Drive Code");
+    private static final ArbitratedFloat leftInput = behaviors.addFloat();
+    private static final ArbitratedFloat rightInput = behaviors.addFloat();
+    private static final Behavior teleop = behaviors.addBehavior("Teleop", FRC.inTeleopMode());
+    private static final Behavior pit = behaviors.addBehavior("Pit Mode", ZukoAzula.mainTuning.getBoolean("Pit Mode Enable", false));
```
Comment (celskeggs) [inline ZukoAzula2016/src/org/team1540/zukoazula/DriveCode.java]: Why is pit mode a tunable value? Also, what happens if we start a match with pit mode enabled?

Comment (AidanSmith) [inline ZukoAzula2016/src/org/team1540/zukoazula/DriveCode.java]: Should it just be a published `BooleanCell`?

Comment (celskeggs) [inline ZukoAzula2016/src/org/team1540/zukoazula/DriveCode.java]: Sure. Make sure to address the other concern as well.

**Comment by celskeggs** [comment]
LGTM!

**Pull request merged**